### PR TITLE
React week3 new branch

### PR DESCRIPTION
**Pull request by elicarcar**
Netlfiy => https://quirky-poincare-334b90.netlify.com/

### CODE REVIEW

File: homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect, useCallback } from "react";
+import UserCard from "../components/UserCard";
+import UserInfo from "../components/UserInfo/UserInfo";
+import Spinner from "../components/Spinner";
+import Button from "../components/Button";
+import "./HomeworkWeek3.css";
+
+function HomeWorkWeek3() {
+  const API = "https://uinames.com/api/?ext";
+  const [data, setData] = useState([]);
```
Comment (ArcoMul) [inline homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js]: `data` is a very generic name, can you find a better describing variable name?

File: homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      const newUserData = await res.json();
+      setIsLoading(false);
+      setData(newUserData);
+    }
+    fetchNewUser();
+
+    return function cleanup() {
+      return fetchNewUser();
+    };
+  }, []);
```
Comment (ArcoMul) [inline homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js]: Cool that you used `useCallback`. This code implements almost the same as on line 13 to 26 though. Try to find a way that you can use either the `useEffect` in both cases or this `useCallback` implementation (to not repeat code unnecessary)

File: homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      async function fetchWithSpace() {
+        setIsLoading(true);
+        const res = await fetch(API);
+        const newUserData = await res.json();
+        setIsLoading(false);
+        setData(newUserData);
+      }
+      return fetchWithSpace();
+    }
+  };
```
Comment (ArcoMul) [inline homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js]: Ah and here is another time almost the same implementation, try to abstract code to a function or to combine the three

**Comment by elicarcar** [comment]
Hello @ArcoMul thank you for your feedback! I implemented all of them, fixed the issue wuth the page jumping around, changed the states name (it's more descriptive now i guess) and instead of writing the code three times, i'm invoking it everytime i need (this is a better solution i guess it keeps the whole code DRY)

**Comment by ArcoMul** [comment]
You can just push your changes in the branch of this pull request. No need to open a new pull request. That way I just losing track of what has been said already and what not :-)

Could you do so? Either merge your new branch in this old one, or look into git cherry picking.

**Pull request closed without merging**